### DISCUSSION TITLE
Laravel fill with enum attribute giver error

### DISCUSSION THREAD

**Discussion by bianchi** [laravel/framework#40557] [Q&A]
I have an Enum like:

    <?php
    
    namespace App\Models\helpers\Enums;
    
    enum ComissionamentoVendedor: string
    {
        case Faturamento = 'No faturamento';
        case Pagamento = 'Após o pagamento pelo cliente';
    }

And I have a model that uses this Enum

    ...
        public $casts = [
            'comissionamento_momento' => ComissionamentoVendedor::class,
        ];
    ...

But when I try to save it I get an error:

    $vendedor = new Vendedor;
    $vendedor->fill($atributos);
    $vendedor->save();

The error is `ErrorException: Attempt to read property "value" on string`

It happens on the file `vendor/illuminate/database/Eloquent/Concerns/HasAttributes.php:951`

     protected function setEnumCastableAttribute($key, $value)
     {
         // $value here is a string, a valid value from ComissionamentoVendedor Enum, sent by the frontend but looks like Laravel expects to be an Enum
         $this->attributes[$key] = isset($value) ? $value->value : null;
     }

```
$atributos contains the values that will fill the attributes of $vendedor. It comes from frontend (React). For example: 
[
 "comissionamento_momento" => "Após o pagamento pelo cliente", // as a string, I expected Lumen to do the conversion to Enum
 "data_nascimento" => null,
 "tipo" => "Física",
 "genero" => "Feminino", 
 "nome" => "João da Silva"
]

Am I doing something wrong? I'm actually using Lumen 8.3.3.

**Comment by henzeb** [comment]
try 8.3.4 first. or if possible, move up to version 9. if no avail, I think you should ask this at the lumen framework github page.
